packages/calcite-components/src/components/table/table.e2e.ts

```diff
@@ -2739,4 +2740,52 @@ describe("keyboard navigation", () => {
       await page.$eval(`#${row3.id}`, (el) => el.shadowRoot?.activeElement.shadowRoot?.querySelector("td").classList),
     ).toEqual({ "0": CSS.selectionCell });
   });
+
+  it("updates table-row's selected property correctly when calciteTabRowSelect event is emitted", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      html`<calcite-table caption="Simple table" selection-mode="multiple">
+        <calcite-table-row slot="table-header">
+          <calcite-table-header heading="Heading" description="Description"></calcite-table-header>
+          <calcite-table-header heading="Heading" description="Description"></calcite-table-header>
+        </calcite-table-row>
+        <calcite-table-row id="row-1">
+          <calcite-table-cell>row1</calcite-table-cell>
+          <calcite-table-cell>row1</calcite-table-cell>
+        </calcite-table-row>
+        <calcite-table-row id="row-2">
+          <calcite-table-cell>row2</calcite-table-cell>
+          <calcite-table-cell>row2</calcite-table-cell>
+        </calcite-table-row>
+      </calcite-table>`,
+    );
+
+    const row1 = await page.find("calcite-table-row");
+    expect(await row1.getProperty("selected")).toBe(false);
+    type TestWindow = typeof window & { selectedValue: boolean };
+
+    await page.evaluate(() => {
```

Comment:
I _think_ you could use the [`createEventTimePropValuesAsserter`](https://github.com/Esri/calcite-design-system/blob/34e3f9bb80a22a13b3946aabc69140d8b9c7619f/packages/calcite-components/src/tests/utils/puppeteer.ts#L592) test util for this.